src/storage/file_manager.c

```diff
@@ -919,7 +925,10 @@ file_header_sanity_check (FILE_HEADER * fhead)
     }
   else
     {
+      int part_cnt = 0;
       assert (!file_extdata_is_empty (part_table) || !VPID_ISNULL (&part_table->vpid_next));
+      (void) file_extdata_all_item_count (thread_p, part_table, &part_cnt);
```

Comment:
please handle error case. the thread might be interrupted, in which case the number will be broken. Only check the count matches if file_extdata_all_item_count returns NO_ERROR. If error is returned, just give up sanity check.